scripts/places_missing_provider_data.py

```diff
@@ -69,3 +69,42 @@ def _is_cache_missing_required_keys(required_keys, cache_for_place_id):
         factual_child = cache_for_place_id['factual']
         if _KEY_WEBSITE not in factual_child: return True
     return False
+
+import pprint
+def calculate_crawled_provider_stats(center, radius_km):
+    statusTable = _firebase.database().child(venuesTable).child("status").get().val()
+
+    # Fetch placeIDs to check
+    location_table = _firebase.database().child(locationsTable).get().val()
+    placeIDs = geo.get_place_ids_in_radius(center, radius_km, location_table)
+
+    print("{} total places found".format(len(placeIDs)))
+
+    provider_dict = { "match": {},
+                      "no_match": {},
+                      "error": {} }
+
+    for placeID in placeIDs:
+        placeStatus = statusTable[placeID]
+        for provider in placeStatus:
+            if provider == "identifiers":
+                continue
+            status = placeStatus[provider]
+            state = ""
+            if status == 0:
+                state = "no_match"
+            elif status == -1:
+                state = "error"
+            else:
+                state = "match"
+            state_dict = provider_dict[state]
+            if provider not in state_dict:
+                state_dict[provider] = 0
+            state_dict[provider] += 1
+
+    pprint.pprint(provider_dict)
+
+if __name__ == '__main__':
+    chicago_center = (41.8338725,-87.688585)
```

Comment:
Passing thought: maybe we should keep a file with random GPS locations that are useful to us (e.g. 4 chicago test locations, ferry building, etc.)